agcaballero/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -73,15 +76,39 @@ public String toString() {
     /**
      * @return a Comment as a HTML div with proper formatting to be displayed
      */
-    private String htmlFormat() {
+    private String htmlFormat(String languageCode) {
       // gets time in local time zone (default: US ET)
       LocalDateTime localTime =
           new LocalDateTime(timestamp.getTime(), DateTimeZone.forID("US/Eastern"));
       DateTimeFormatter formatter = DateTimeFormat.forPattern("h:mm a M/dd/yy");
 
       return "<div class='comment-div'>"
           + "<p class='date'>" + formatter.print(localTime) + "</p>"
-          + "<p><b>" + name + " (" + email + "):</b> <br><br>" + text + "</p></div>";
+          + "<p><b>" + name + " (<a href='mailto:" + email + "'>" + email + "</a>):</b></p>"
+          + "<p class='comment-text' lang=" + languageCode + ">" + getTranslatedText(languageCode) + "</p></div>";
+    }
+
+    /**
+     * @return the text field translated to the language of a given language code (e.g. "EN" for English)
+     * [Makes a network call to Translation API so method could be slow]
+     */
+    public String getTranslatedText(String languageCode) {
+      // declare an instance of translate
+      Translate translate = TranslateOptions.getDefaultInstance().getService();
+
+      // detect the current language of text
+      String currentLanguage = translate.detect(text).getLanguage();
```

Comment:
Seems like this will also cost money, and could be avoided by caching the current language code?

Reply:
This makes sense. I'm planning to store which language it was submitted in via dropdown. Then, not translate it if this is same as current language.

Reply:
Done